src/engine/database/gremlin.rs

```diff
@@ -431,12 +431,12 @@ impl GremlinTransaction {
 }
 
 #[async_trait]
-impl Transaction for GremlinTransaction {
+impl<RequestCtx: RequestContext> Transaction<RequestCtx> for GremlinTransaction {
```

Comment:
This came up when passing a `Transaction` to the event handler. The only way I could get it to compile was to move up the type parameter and then I had to propagate it all over the place. I definitely tried to avoid it - so if there is a way to do away with it that you can spot, I would be game for reverting it.

Reply:
Right now the `Transaction` reference in the `EventFacade` in a `transaction: &'a mut dyn Transaction<RequestCtx>`.  If you make that a `transaction: &'a mut T` where `T: Transaction`, does that free you up from some of the constraints, including the need to put the `RequestCtx` in the header here rather than on individual methods?

Reply:
I had no luck. If you find yourself with a few minutes, could you take a stab at it?